common/src/types/v0/store/node.rs

```diff
@@ -66,33 +165,242 @@ impl NodeSpec {
     pub fn labels(&self) -> &NodeLabels {
         &self.labels
     }
+    /// Node labels
+    pub fn drain_state(&self) -> &Option<DrainState> {
+        &self.drain_state
+    }
+
     /// Node gRPC endpoint
     pub fn set_endpoint(&mut self, endpoint: String) {
         self.endpoint = endpoint
     }
+
     /// Cordon node by applying the label.
     pub fn cordon(&mut self, label: String) {
-        self.cordon_labels.push(label);
+        // the the node has the label, return with an error
+        match self.drain_state.clone() {
```

Comment:
nit: I think this can be made simpler by getting a mutable reference  and changing it, rather than resetting the drain_state in its entirety if add_cordon_label takes a mut reference instead of moving.

Reply:
Changed to updating the state instead of replacing, wherever the enum doesn't change, here and in other places.